### PR TITLE
DLFS - A filesystem built on replicated logs

### PR DESCRIPTION
**Pull request by gerritsundaram**
- implement the hadoop FileSystem api

### CODE REVIEW

**Comment by jiazhai** [comment]
Is there a test case for it? 
So we would know a little of the use-case for DLFS.
:)

**Comment by sijie** [comment]
@gerritsundaram any updates?

File: distributedlog-contribs/distributedlog-hdfs/src/main/java/com/twitter/distributedlog/fs/DistributedLogFileSystem.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    @Override
+    public FSDataOutputStream create(Path path,
+                                     FsPermission fsPermission,
+                                     boolean overwrite,
+                                     int bufferSize,
+                                     short replication,
+                                     long blockSize,
+                                     Progressable progressable) throws IOException {
+
+        // TODO: support overwrite, support permission
```
Comment (jiazhai) [inline distributedlog-contribs/distributedlog-hdfs/src/main/java/com/twitter/distributedlog/fs/DistributedLogFileSystem.java]: Shall we open a github issue for this?

File: distributedlog-contribs/distributedlog-hdfs/src/main/java/com/twitter/distributedlog/fs/DistributedLogOutputStream.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public void write(byte[] b) throws IOException {
+        streamWriter.write(b);
+    }
+
+    @Override
+    public void write(byte[] b, int off, int len) throws IOException {
+        // TODO: improve AppendOnlyStreamWriter
```
Comment (jiazhai) [inline distributedlog-contribs/distributedlog-hdfs/src/main/java/com/twitter/distributedlog/fs/DistributedLogOutputStream.java]: Shall we open a github issue for this "todo"?

**Comment by sijie** [comment]
Close this since it is overtaken by #227. marked it as duplicated.

**Pull request closed without merging**